### ISSUE TITLE
Crash because of missing proguard rule?

### ISSUE THREAD

**Issue by Alexs784** [Kotlin/kotlinx.coroutines#2192]
### (partial) stack trace:
```
java.lang.IllegalStateException: Module with Main dispatcher is missing.
Add dependency with required Main dispatcher, e.g. 'kotlinx-coroutines-android'
```

### Description
I am having production builds (=with `R8` enabled) crashing after updating from `1.3.7` to `1.3.8`.
Adding the pro guard rule `-keep class kotlinx.coroutines.android.AndroidDispatcherFactory {*;}`
_solves_ the issue.

Is that the real fix or is there something else?

Full proguard rules regarding coroutines:
```
#--------------- Coroutines ---------------
-keep,allowobfuscation @interface kotlin.coroutines.jvm.internal.** { *; }
-keep class kotlinx.coroutines.android.AndroidDispatcherFactory {*;}
-keepnames class kotlinx.coroutines.internal.MainDispatcherFactory {}
-keepnames class kotlinx.coroutines.CoroutineExceptionHandler {}
-keepclassmembernames class kotlinx.** {
    volatile <fields>;
}
```

**Comment by qwwdfsad** [comment]
That's the fix, indeed.
The root cause seems to be https://github.com/Kotlin/kotlinx.coroutines/commit/3a8a0ea419f98ebc4534c11d96c9dd91f0a10f2c and it is fixed with  3cc9b94f8f9ce68d99d46120ee48e6f6ff61c160

Could you please check whether you have the same problem with kotlinx.coroutines 1.3.9?

**Comment by Alexs784** [comment]
I can confirm it works properly with the `1.3.9` version, removing the rule:
`-keep class kotlinx.coroutines.android.AndroidDispatcherFactory {*;}`

So basically proguard rules can be reverted to the old ones:
```
#--------------- Coroutines ---------------
-keep,allowobfuscation @interface kotlin.coroutines.jvm.internal.** { *; }
-keepnames class kotlinx.coroutines.internal.MainDispatcherFactory {}
-keepnames class kotlinx.coroutines.CoroutineExceptionHandler {}
-keepclassmembernames class kotlinx.** {
    volatile <fields>;
}
```

**Comment by qwwdfsad** [comment]
Fixed in 1.3.9
